src/language-js/print/array.js

```diff
@@ -106,6 +98,16 @@ function printArray(path, options, print) {
           )
       );
 
+    const trailingComma = !canHaveTrailingComma
+      ? ""
+      : needsForcedTrailingComma
+      ? ","
+      : !shouldPrintComma(options)
+      ? ""
+      : ifBreak(",", "", {
+          groupId: shouldUseConciseFormatting ? groupId : undefined,
+        });
```

Comment:
```suggestion
let trailingComma = "";
if (canHaveTrailingComma) {
  if (needsForcedTrailingComma) {
    trailingComma = ",";
  } else if (shouldPrintComma(options)) {
    trailingComma = ifBreak(",", "", {
      groupId: shouldUseConciseFormatting ? groupId : undefined,
    });
  }
}
```

Reply:
My version looks better to me :) Maybe it's better to extract `shouldUseConciseFormatting` to the outer conditional chain though.

Reply:
It reads like a chain of simple single-line yes/no questions. I see nothing wrong with it.

Reply:
Honestly, I never used a ternary nested more than 2 levels, so I never care about https://github.com/prettier/prettier/pull/9559

I can read it, but a little slow than `if`.

Reply:
Another thing I don't like in your version, there are two negated conditions, I usually avoid it.

Reply:
When I see nested `if`s like that, it gives me an impression that something tricky and intimidating (=== "slower to read") is going on there that can't be expressed as a simple yes/no chain. That said, I prefer nested `if`s when conditions are more complex than in this case. I really appreciate that you're a fan of ESLint (thanks for all those internal rules), but IMHO you're too strict on innocent code structures that are perfectly okay.